model/task_lifecycle.go

```diff
@@ -796,11 +796,31 @@ func evalStepback(t *task.Task, caller, status string, deactivatePrevious bool)
 		if err != nil {
 			return errors.WithStack(err)
 		}
-		if shouldStepBack {
-			if err = doStepback(t); err != nil {
-				return errors.Wrap(err, "Error during step back")
+		if !shouldStepBack {
+			return nil
+		}
+
+		if t.IsPartOfSingleHostTaskGroup() {
+			// stepback earlier task group tasks as well
+			catcher := grip.NewBasicCatcher()
+			tasks, err := task.FindTaskGroupFromBuild(t.BuildId, t.TaskGroup)
+			if err != nil {
+				return errors.Wrapf(err, "can't get task group for task '%s'", t.Id)
+			}
+			if len(tasks) == 0 {
+				return errors.Errorf("no tasks in task group '%s' for task '%s'", t.TaskGroup, t.Id)
 			}
+			for _, tgTask := range tasks {
+				catcher.Wrapf(doStepback(&tgTask), "error stepping back task group task '%s'", tgTask.DisplayName)
+				if tgTask.Id == t.Id {
+					break // don't need to stepback later tasks in the group
```

Comment:
are we certain that they are stored in this order? with the earliest being the first in the array and the latest being the last in the array? (I'm guessing yes considering the tests pass...)

Reply:
Yep: task groups are defined and stored in this way.